dulwich/porcelain.py

```diff
@@ -3307,6 +3307,35 @@ def get_commit_date(branch_name: bytes) -> int:
         return branches
 
 
+def merged_branches(repo: RepoPath) -> Iterator[bytes]:
+    """List branches that have been merged into the current branch.
+
+    Args:
+      repo: Path to the repository
+    Yields:
+      Branch names (without refs/heads/ prefix) that are merged
+      into the current HEAD
+    """
+    with open_repo_closing(repo) as r:
+        current_sha = r.refs[b"HEAD"]
+
+        for branch_ref in r.refs.keys(base=b"refs/heads/"):
+            full_ref = b"refs/heads/" + branch_ref
+
+            if full_ref not in r.refs:
+                continue
+
+            try:
+                branch_sha = r.refs[full_ref]
+
+                # Check if branch is an ancestor of HEAD (fully merged)
+                if can_fast_forward(r, branch_sha, current_sha):
+                    yield branch_ref
+
+            except (KeyError, ValueError):
```

Comment:
same here - .keys() yielded this ref so it should exist. Why would KeyError or ValueError be raised?